src/AppInstallerCLICore/Workflows/UpdateFlow.cpp

```diff
@@ -176,6 +179,18 @@ namespace AppInstaller::CLI::Workflow
                 continue;
             }
 
+            // Filter out packages that require explicit upgrades.
+            // Note that this is filtering based on the updated installer, not the installed version. So there

```

Comment:
We could store this information in the tracking catalog so that we can make the decision based on the "current" version.  I envisioned that it is the same location we would store the pinning data, which is basically a user driven version of this feature.

There will always be problems if the state changes, whether it be an update to the current version or a newer version changing it.  We could potentially solve it by not even storing the value but always looking it up from the manifest for the "most likely version" (something like "greatest version that is less than or equal to the current version").

I think either of these options is better than what you have, which will guarantee a gap in the experience if the state changes across versions.